plugins/modules/zos_zfs_resize.py

```diff
@@ -0,0 +1,718 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) IBM Corporation 2025
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+module: zos_zfs_resize
+version_added: '1.13.0'
+short_description: Resize a zfs data set.
+description:
+  - The module L(zos_zfs_resize.,/zos_zfs_resize.html) can resize a zFS aggregate data set.
+  - The I(target) data set must be a unique and a Fully Qualified Name (FQN) of a 1-OS zFS aggregate data set.
+  - The data set must be attached as read-write, and contain only one operating system.
+  - I(size) must be provided.
+author:
+  - "Rich Parker (@richp405)"
+  - "Marcel Gutierrez (@AndreMarcel99)"
+options:
+  target:
```

Comment:
I think it would be good to discuss this with the team, I know its coming late but have you thought about using `src` instead of `target` for the sake of consistency with other modules. I am ok with what everyone decides, its just we have never used `target` and while it makes sense, we might want to consider if `src` makes sense as well , otherwise we can leave it as is.